### DISCUSSION TITLE
two-color syntax highlighting (vs code has more colors)

### DISCUSSION THREAD

**Discussion by ivang7** [coder/code-server#2278] [Q&A]
Hey.

I noticed that in the code-server only three colors are used for syntax highlighting (blue for keywords, brown for strings, white for the rest).

VS code uses a lot more colors to highlight different elements - classes, interfaces, methods, etc.

Can this be fixed? maybe some extensions? or does it need to be improved on the code-server?
Are there any plans to make the highlighting more informative, similar to vs code?
![Code_CctxqsDsoI](https://user-images.githubusercontent.com/10797493/98453324-ed827f80-2168-11eb-9036-687eb5cf27bf.png)

**Comment by code-asher** [comment]
I wonder if it's because code-server is using an older version of VS
Code? code-server 3.6.2 is using VS Code 1.50.0 for example and 1.51.0
was just recently released.

What versions of code-server and VS Code are you using? And do you have
all the same extensions and themes installed on both?

**Reply by ivang7** [reply]
I apologize for not answering for a long time.
The theme and plugins are installed the same.

Possibly the difference in LanguageServer for C #

When looking at the token and scope in the code-sever, you can see that there is no specific definition for the interface
code-server:
![code-server](https://user-images.githubusercontent.com/74328378/99156646-98e48480-26d3-11eb-82b7-63c0056ecaa5.png)

visual code:
![visual-code](https://user-images.githubusercontent.com/74328378/99156656-abf75480-26d3-11eb-8e7a-a630f9a29600.png)

**Reply by code-asher** [reply]
What versions of code-server and VS Code are you using?

I tried running the latest VS Code insiders with the `Dark+` theme and created a new project with `dotnet new` but when I hover over the same token I only see `storage.type.cs source.cs`; I don't get anything about interfaces. I also tried the latest non-insiders VS Code but had the same result.

The only extension I have installed is the C# extension. I'm getting intellisense so the language server appears to be running. Do you have any other C#-related extensions installed? Maybe I'm missing one.

It looks like in your latest screenshot though code-server now has all the highlighting it was missing in your first screenshot. Was code-server initially using the `Dark` theme? That theme seems to have fewer colors.

**Reply by code-asher** [reply]
Screenshot for reference, apologies for the bad quality as it's over VNC. This is from VS Code 1.51.1 on Ubuntu with dotnet and the C# extension installed using the `Dark+` theme.

![vscode](https://user-images.githubusercontent.com/45609798/99295847-879e9380-280b-11eb-9434-cc1b44b29337.png)
